src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c

```diff
@@ -29,23 +30,140 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
+                           this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z,
+                           OBJTOKEITOBIRA_PARAM(OBJTOKEITOBIRA_TYPE_1));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Update.s")
+    if ((type == OBJTOKEITOBIRA_TYPE_0) && !(CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04)) &&
+        (play->sceneId == SCENE_CLOCKTOWER) && (gSaveContext.sceneLayer == 0) && (this->dyna.actor.csId >= 0)) {
+        this->dyna.actor.flags |= ACTOR_FLAG_10;
+        this->actionFunc = ObjTokeiTobira_StartCutscene;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Draw.s")
+void ObjTokeiTobira_Destroy(Actor* thisx, PlayState* play) {
+    ObjTokeiTobira* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjTokeiTobira_StartCutscene(ObjTokeiTobira* this) {
+    if (CutsceneManager_IsNext(this->dyna.actor.csId)) {
+        CutsceneManager_StartWithPlayerCs(this->dyna.actor.csId, &this->dyna.actor);
+        SET_WEEKEVENTREG(WEEKEVENTREG_59_04);
+        this->actionFunc = NULL;
+        this->dyna.actor.flags &= ~ACTOR_FLAG_10;
+    } else {
+        CutsceneManager_Queue(this->dyna.actor.csId);
+    }
+}
+
+void ObjTokeiTobira_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTokeiTobira* this = THIS;
+    Player* player = GET_PLAYER(play);
+    s32 pad50;
+    s32 temp_param = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
+    f32 sp48 = D_80ABD778[temp_param];
+    s32 pad44;
+
+    if (player->actor.bgCheckFlags & BGCHECKFLAG_PLAYER_WALL_INTERACT) {
+        if (DynaPoly_GetActor(&play->colCtx, player->actor.wallBgId) == &this->dyna) {
+            f32 sp40;
+            Vec3f sp34;
+
+            Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &sp34, &player->actor.world.pos);
+            sp40 = sp34.x * sp48;
+
+            if (sp40 > 20.0f) {
+                if (sp34.z > 0.0f) {
+                    this->unk160 += (sp40 - 20.0f) * sp48 * (16.0f / 3.0f);
+                    this->unk168 = 1;
+                    if (this->unk16C <= 0) {
+                        Actor_PlaySfx(&this->dyna.actor, NA_SE_OC_WOOD_GATE_OPEN);
+                        this->unk16C = 0x50;
+                    }
+                }
+
+                if (sp40 > 48.0f) {
+                    ObjTokeiTobira* tobira;
+                    if (temp_param == OBJTOKEITOBIRA_TYPE_0) {
```

Comment:
Leave an empty line between the variable declaration and the rest of the code
```suggestion
                    ObjTokeiTobira* tobira;

                    if (temp_param == OBJTOKEITOBIRA_TYPE_0) {
```